TypeError: np.matrix is not supported with sklearn 1.2.0

Running the testsuite against scikit-learn 1.2.0 (up from 1.1.3)  we (nixpkgs) are seeing 17 failing variations of the `sparse_spartial_check` test:

```
FAILED pynndescent/tests/test_distances.py::test_sparse_spatial_check[euclidean] - TypeError: np.matrix is not supported. Please convert to a numpy array with...
FAILED pynndescent/tests/test_distances.py::test_sparse_spatial_check[manhattan] - TypeError: np.matrix is not supported. Please convert to a numpy array with...
FAILED pynndescent/tests/test_distances.py::test_sparse_spatial_check[chebyshev] - TypeError: np.matrix is not supported. Please convert to a numpy array with...
FAILED pynndescent/tests/test_distances.py::test_sparse_spatial_check[minkowski] - TypeError: np.matrix is not supported. Please convert to a numpy array with...
FAILED pynndescent/tests/test_distances.py::test_sparse_spatial_check[hamming] - TypeError: np.matrix is not supported. Please convert to a numpy array with...
FAILED pynndescent/tests/test_distances.py::test_sparse_spatial_check[canberra] - TypeError: np.matrix is not supported. Please convert to a numpy array with...
FAILED pynndescent/tests/test_distances.py::test_sparse_spatial_check[cosine] - TypeError: np.matrix is not supported. Please convert to a numpy array with...
FAILED pynndescent/tests/test_distances.py::test_sparse_spatial_check[braycurtis] - TypeError: np.matrix is not supported. Please convert to a numpy array with...
FAILED pynndescent/tests/test_distances.py::test_sparse_spatial_check[correlation] - TypeError: np.matrix is not supported. Please convert to a numpy array with...
FAILED pynndescent/tests/test_distances.py::test_sparse_binary_check[jaccard] - TypeError: np.matrix is not supported. Please convert to a numpy array with...
FAILED pynndescent/tests/test_distances.py::test_sparse_binary_check[matching] - TypeError: np.matrix is not supported. Please convert to a numpy array with...
FAILED pynndescent/tests/test_distances.py::test_sparse_binary_check[dice] - TypeError: np.matrix is not supported. Please convert to a numpy array with...
FAILED pynndescent/tests/test_distances.py::test_sparse_binary_check[kulsinski] - TypeError: np.matrix is not supported. Please convert to a numpy array with...
FAILED pynndescent/tests/test_distances.py::test_sparse_binary_check[rogerstanimoto] - TypeError: np.matrix is not supported. Please convert to a numpy array with..
FAILED pynndescent/tests/test_distances.py::test_sparse_binary_check[russellrao] - TypeError: np.matrix is not supported. Please convert to a numpy array with...
FAILED pynndescent/tests/test_distances.py::test_sparse_binary_check[sokalmichener] - TypeError: np.matrix is not supported. Please convert to a numpy array with...
FAILED pynndescent/tests/test_distances.py::test_sparse_binary_check[sokalsneath] - TypeError: np.matrix is not supported. Please convert to a numpy array with...
```

via https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/metrics/pairwise.py#L146
in https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/utils/validation.py#L736-L741

--- a/pynndescent/tests/test_distances.py
+++ b/pynndescent/tests/test_distances.py
@@ -174,7 +174,7 @@ def test_sparse_spatial_check(sparse_spatial_data, metric, decimal=6):
 )
 def test_sparse_binary_check(sparse_binary_data, metric):
     if metric in spdist.sparse_named_distances:
-        dist_matrix = pairwise_distances(sparse_binary_data.todense(), metric=metric)
+        dist_matrix = pairwise_distances(np.asarray(sparse_binary_data.todense()), metric=metric)
     if metric in ("jaccard", "dice", "sokalsneath"):
         dist_matrix[np.where(~np.isfinite(dist_matrix))] = 0.0
     if metric in ("kulsinski", "russellrao"):

--- a/pynndescent/tests/test_distances.py
+++ b/pynndescent/tests/test_distances.py
@@ -109,7 +109,7 @@ def test_binary_check(binary_data, metric):
 def test_sparse_spatial_check(sparse_spatial_data, metric, decimal=6):
     if metric in spdist.sparse_named_distances:
         dist_matrix = pairwise_distances(
-            sparse_spatial_data.todense().astype(np.float32), metric=metric
+            np.asarray(sparse_spatial_data.todense()).astype(np.float32), metric=metric
         )
     if metric in ("braycurtis", "dice", "sokalsneath", "yule"):
         dist_matrix[np.where(~np.isfinite(dist_matrix))] = 0.0

